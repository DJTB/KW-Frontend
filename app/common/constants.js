/* eslint-disable prefer-destructuring */
// https://github.com/mrsteele/dotenv-webpack/issues/70
export const VERSION = process.env.VERSION;
window.KW_VERSION = VERSION;
export const RAVEN_DSN = process.env.RAVEN_DSN;
const DEV_DOMAIN = process.env.DEV_DOMAIN;
const PROD_DOMAIN = 'https://www.kaniwani.com';
export const IS_DEV_ENV = process.env.NODE_ENV === 'development';
export const IS_PROD_ENV = process.env.NODE_ENV === 'production';
const BASE_URL = IS_DEV_ENV ? DEV_DOMAIN : PROD_DOMAIN;
/* eslint-enable prefer-destructuring */

export const API_BASE = `${BASE_URL}/api/v1/`;

export const DATE_FORMAT = 'MMM Do YYYY';
export const TIME_FORMAT = 'hh:mma';
export const DATE_TIME_FORMAT = `${DATE_FORMAT}[,] ${TIME_FORMAT}`;

export const MAX_NOTES_LENGTH = 500;
export const MAX_CONTACT_LENGTH = 1000;

export const WK_SRS_RANKS = {
  ONE: 'APPRENTICE',
  TWO: 'GURU',
  THREE: 'MASTER',
  FOUR: 'ENLIGHTENED',
  FIVE: 'BURNED',
};

export const SRS_RANKS = {
  ZERO: 'UNTRAINED',
  ...WK_SRS_RANKS,
};

export const SRS_RANGES = {
  ZERO: [0],
  ONE: [1, 2, 3, 4],
  TWO: [5, 6],
  THREE: [7],
  FOUR: [8],
  FIVE: [9],
};

export const KEYCODES = {
  SPACE: 32,
  ENTER: 13,
};

export const TAGS = {
  common: 'common',
  MA: 'Martial Arts',
  X: 'Rude/X-Rated',
  abbr: 'Abbreviation',
  adj: 'Adjective',
  'adj-i': 'I Adjective',
  'adj-ix': 'I Adjective',
  'adj-na': 'Na Adjective',
  'adj-no': 'No Adjective',
  'adj-pn': 'Pre-Noun Adjectival',
  'adj-t': 'Taru Adjective',
  'adj-f': 'Prenominal Noun/Verb',
  adv: 'Adverb',
  'adv-to': 'To Adverb',
  arch: 'Archaism',
  ateji: 'Ateji',
  aux: 'Auxiliary',
  'aux-v': 'Auxiliary Verb',
  'aux-adj': 'Auxiliary Adjective',
  Buddh: 'Buddhist',
  chem: 'Chemistry',
  chn: 'Childish',
  col: 'Colloquial',
  comp: 'Computer',
  conj: 'Conjunction',
  'cop-da': 'Copula',
  ctr: 'Counter',
  derog: 'Derogatory',
  eK: 'Kanji Only',
  ek: 'Kana Only',
  exp: 'Expression',
  fam: 'Familiar',
  fem: 'Feminine',
  food: 'Food',
  geom: 'Geometry',
  gikun: 'Gikun/Jukujikun',
  hon: 'Honorific (sonkeigo)',
  hum: 'Humble (kenjougo)',
  iK: 'Irregular Kanji',
  id: 'Idiom',
  ik: 'Irregular Kana',
  int: 'Interjection',
  io: 'Irregular Okurigana',
  iv: 'Irregular Verb',
  ling: 'Linguistics',
  'm-sl': 'Manga Slang',
  male: 'Masculine',
  'male-sl': 'Male Slang',
  math: 'Mathematics',
  mil: 'Military',
  n: 'Noun',
  'n-adv': 'Adverbial Noun',
  'n-suf': 'Noun Suffix',
  'n-pref': 'Noun Prefix',
  'n-t': 'Temporal Noun',
  num: 'Numeric',
  oK: 'Out-dated Kanji',
  obs: 'Obsolete',
  obsc: 'Obscure',
  ok: 'Out-Dated Kana',
  oik: 'Old/Irregular Kana',
  'on-mim': 'Onomatopoeic/Mimetic',
  pn: 'Pronoun',
  poet: 'Poetry',
  pol: 'Polite (teineigo)',
  pref: 'Prefix',
  proverb: 'Proverb',
  prt: 'Particle',
  physics: 'Physics',
  rare: 'Rare',
  sens: 'Sensitive',
  sl: 'Slang',
  suf: 'Suffix',
  uK: 'Usually Kanji',
  uk: 'Usually Kana',
  unc: 'Unclassified',
  yoji: 'Yojijukugo',
  v1: 'Ichidan Verb',
  'v1-s': 'Ichidan Verb',
  'v2a-s': 'Nidan Verb',
  v4h: 'Yodan Verb',
  v4r: 'Yodan Verb',
  v5: 'Godan Verb',
  v5aru: 'Godan Verb',
  v5b: 'Godan Verb',
  v5g: 'Godan Verb',
  v5k: 'Godan Verb',
  'v5k-s': 'Godan Verb',
  v5m: 'Godan Verb',
  v5n: 'Godan Verb',
  v5r: 'Godan Verb',
  'v5r-i': 'Godan Verb',
  v5s: 'Godan Verb',
  v5t: 'Godan Verb',
  v5u: 'Godan Verb',
  'v5u-s': 'Godan Verb',
  v5uru: 'Godan Verb',
  vz: 'Ichidan Verb',
  vi: 'Intransitive Verb',
  vk: 'Kuru Verb',
  vn: 'Irregular Verb',
  vr: 'Irregular Verb',
  vs: 'Suru Verb',
  'vs-c': 'Su Verb',
  'vs-s': 'Suru Verb',
  'vs-i': 'Irregular Suru Verb',
  kyb: 'Kyoto-ben',
  osb: 'Osaka-ben',
  ksb: 'Kansai-ben',
  ktb: 'Kantou-ben',
  tsb: 'Tosa-ben',
  thb: 'Touhoku-ben',
  tsug: 'Tsugaru-ben',
  kyu: 'Kyuushuu-ben',
  rkb: 'Ryuukyuu-ben',
  nab: 'Nagano-ben',
  hob: 'Hokkaido-ben',
  vt: 'Transitive verb',
  vulg: 'Vulgar',
  'adj-kari': 'Adjective',
  'adj-ku': 'Adjective',
  'adj-shiku': 'Adjective',
  'adj-nari': 'Adjective',
  'n-pr': 'Proper Noun',
  'v-unspec': 'Verb',
  v4k: 'Yodan Verb',
  v4g: 'Yodan Verb',
  v4s: 'Yodan Verb',
  v4t: 'Yodan Verb',
  v4n: 'Yodan Verb',
  v4b: 'Yodan Verb',
  v4m: 'Yodan Verb',
  'v2k-k': 'Nidan Verb',
  'v2g-k': 'Nidan Verb',
  'v2t-k': 'Nidan Verb',
  'v2d-k': 'Nidan Verb',
  'v2h-k': 'Nidan Verb',
  'v2b-k': 'Nidan Verb',
  'v2m-k': 'Nidan Verb',
  'v2y-k': 'Nidan Verb',
  'v2r-k': 'Nidan Verb',
  'v2k-s': 'Nidan Verb',
  'v2g-s': 'Nidan Verb',
  'v2s-s': 'Nidan Verb',
  'v2z-s': 'Nidan Verb',
  'v2t-s': 'Nidan Verb',
  'v2d-s': 'Nidan Verb',
  'v2n-s': 'Nidan Verb',
  'v2h-s': 'Nidan Verb',
  'v2b-s': 'Nidan Verb',
  'v2m-s': 'Nidan Verb',
  'v2y-s': 'Nidan Verb',
  'v2r-s': 'Nidan Verb',
  'v2w-s': 'Nidan Verb',
  archit: 'Architecture',
  astron: 'Astronomy',
  baseb: 'Baseball',
  biol: 'Biology',
  bot: 'Botany',
  bus: 'Business',
  econ: 'Economics',
  engr: 'Engineering',
  finc: 'Finance',
  geol: 'Geology',
  law: 'Law',
  mahj: 'Mahjong',
  med: 'Medicine',
  music: 'Music',
  Shinto: 'Shinto',
  shogi: 'Shogi',
  sports: 'Sports',
  sumo: 'Sumo',
  zool: 'Zoology',
  joc: 'Joking',
  anat: 'Anatomical',
};
