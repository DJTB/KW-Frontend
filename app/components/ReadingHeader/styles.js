import styled from 'styled-components';

import H3 from 'base/H3';
import A from 'base/A';
import TagsList from 'components/TagsList';

import { elementGutter } from 'shared/styles/layout';
import { transparent, whiteLight, purple } from 'shared/styles/colors';
import { outerLine } from 'shared/styles/shadows';

export const Wrapper = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
`;

export const Heading = styled(H3)`
  ${elementGutter}
  display: flex;
  margin: 0;
  opacity: .8;
  text-align: left;
  flex: 0 0 auto;
  align-self: flex-start;
`;

export const VocabLink = styled(A)`
  text-decoration: none;
  color: ${whiteLight};
  border-radius: 5px;
  background-color: ${purple};
  margin: 0 .15em;
  align-self: flex-end;
  font-size: .7em;
  line-height: 1.1;
  padding: .25em .5em;
  opacity: .9;

  &:first-child {
    margin-left: .75em;
  }
  &:hover {
    color: ${purple};
    background-color: ${transparent};
    box-shadow: ${outerLine};
  }
`;

export const Tags = styled(TagsList)`
  ${elementGutter}
  padding-left: 1rem;
  display: flex;
  flex-flow: row wrap;
  flex: 1 1 0%;
  align-self: flex-end;
  justify-content: flex-end;
  /* negate chip margin */
  margin-top: -.15em;
  margin-bottom: -.15em;
`;
