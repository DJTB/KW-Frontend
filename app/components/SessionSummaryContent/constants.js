export const DEFAULT_ACTION = 'app/SessionSummaryPage/DEFAULT_ACTION';
