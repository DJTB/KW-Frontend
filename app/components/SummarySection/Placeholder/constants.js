export const PLACEHOLDERS = {
  CORRECT: {
    text: ' 満点！',
    kaomoji: ' (๑•̀ㅂ•́)و',
  },
  INCORRECT: {
    text: '零点',
    kaomoji: 'ʕノ•ᴥ•ʔノ ︵ ┻━┻',
  },
  CRITICAL: {
    text: '無事！',
    kaomoji: '(^・ω・^ )',
  },
};
