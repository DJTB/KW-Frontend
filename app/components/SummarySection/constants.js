export const CRITICAL = 'CRITICAL';
export const CORRECT = 'CORRECT';
export const INCORRECT = 'INCORRECT';

export const TYPES = {
  [CORRECT]: {
    color: 'green',
  },
  [INCORRECT]: {
    color: 'red',
  },
  [CRITICAL]: {
    color: 'orange',
  },
};
