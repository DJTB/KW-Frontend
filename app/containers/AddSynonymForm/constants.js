 export const LOAD_JISHODATA = 'kw/AddSynonymForm/LOAD_JISHODATA';
 export const LOAD_JISHODATA_SUCCESS = 'kw/AddSynonymForm/LOAD_JISHODATA_SUCCESS';
 export const LOAD_JISHODATA_ERROR = 'kw/AddSynonymForm/LOAD_JISHODATA_ERROR';

 export const ADD_SYNONYM = 'kw/AddSynonymForm/ADD_SYNONYM';
 export const ADD_SYNONYM_SUCCESS = 'kw/AddSynonymForm/ADD_SYNONYM_SUCCESS';
 export const ADD_SYNONYM_ERROR = 'kw/AddSynonymForm/ADD_SYNONYM_ERROR';
 export const REMOVE_SYNONYM = 'kw/AddSynonymForm/REMOVE_SYNONYM';
 export const REMOVE_SYNONYM_ERROR = 'kw/AddSynonymForm/REMOVE_SYNONYM_ERROR';
