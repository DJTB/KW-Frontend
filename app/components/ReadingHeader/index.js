import React from 'react';
import PropTypes from 'prop-types';

import { Wrapper, Heading, VocabLink } from './styles';

ReadingHeader.propTypes = {
  character: PropTypes.string.isRequired,
  useAlcPro: PropTypes.bool.isRequired,
};

const createAlcLink = (character, useAlcPro) =>
  `http://${useAlcPro ? 'eowpf' : 'eow'}.alc.co.jp/search?q=${encodeURIComponent(character)}`;
const createGooLink = (character) => `http://dictionary.goo.ne.jp/srch/all/${encodeURIComponent(character)}/m0u/`;
const createWeblioLink = (character) => `http://ejje.weblio.jp/content/${encodeURIComponent(character)}`;
const createForvoLink = (character) => `http://forvo.com/search/${encodeURIComponent(character)}/`;

function ReadingHeader({ character, useAlcPro }) {
  return (
    <Wrapper>
      <Heading>
        Reading
      </Heading>
      <VocabLink
        href={`//wanikani.com/vocabulary/${character}`}
        title="View on WaniKani"
        external
      >
        WK
      </VocabLink>
      <VocabLink
        href={createAlcLink(character, useAlcPro)}
        title="View on Alc (Eijiro)"
        external
      >
        ALC
      </VocabLink>
      <VocabLink
        href={createGooLink(character)}
        title="View on Goo"
        external
      >
        GOO
      </VocabLink>
      <VocabLink
        href={createWeblioLink(character)}
        title="View on Weblio"
        external
      >
        WEBLIO
      </VocabLink>
      <VocabLink
        href={createForvoLink(character)}
        title="View on Forvo"
        external
      >
        FORVO
      </VocabLink>
    </Wrapper>
  );
}

export default ReadingHeader;
