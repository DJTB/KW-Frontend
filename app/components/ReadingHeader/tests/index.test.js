import React from 'react';
import { shallow } from 'enzyme';
import { PARTS_OF_SPEECH } from 'shared/constants';
import ReadingHeader from '../index';

describe('<ReadingHeader />', () => {
  it('should match baseline snapshot', () => {
    const renderedComponent = shallow(
      <ReadingHeader
        wkVocabLink="#"
        kwVocabLink="#"
        tags={PARTS_OF_SPEECH.slice(0, 10)}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
