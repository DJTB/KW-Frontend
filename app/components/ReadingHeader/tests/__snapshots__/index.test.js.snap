// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReadingHeader /> should match baseline snapshot 1`] = `
<styled.div>
  <styled.h3>
    Reading
  </styled.h3>
  <Styled(A)
    external={true}
    href="#"
    title="View on WaniKani"
  >
    WK
  </Styled(A)>
  <Styled(A)
    target="_blank"
    title="View on KaniWani"
    to="#"
  >
    KW
  </Styled(A)>
  <Styled(TagsList)
    tags={
      Array [
        "JLPT N1",
        "JLPT N2",
        "JLPT N3",
        "JLPT N4",
        "JLPT N5",
        "Common",
        "Intransitive verb",
        "Godan verb",
        "Noun",
        "Suru verb",
      ]
    }
  />
</styled.div>
`;
